boards/arm/tiva/lm3s6965-ek/src/lm_appinit.c

```diff
@@ -152,6 +153,9 @@ int board_app_initialize(uintptr_t arg)
 
   mcinfo("Successfully bound SPI port %d to MMC/SD slot %d\n",
          CONFIG_NSH_MMCSDSPIPORTNO, CONFIG_NSH_MMCSDSLOTNO);
+#endif
```

Comment:
I am not going to hold this up any longer.  But this means that the logic about the #ifndef CONFIG_BOARD_LATE_INITIALIZE will not be execute when initialization is done via board_late_initialize.  I don't think you really want that asymmetry.

Reply:
how should it be?
i tried to follow what sim and others do.

Reply:
Look for an example at:
https://github.com/apache/incubator-nuttx/blob/master/boards/arm/stm32/stm32f4discovery/src/stm32_boot.c#L124
https://github.com/apache/incubator-nuttx/blob/master/boards/arm/stm32/stm32f4discovery/src/stm32_appinit.c#L58
https://github.com/apache/incubator-nuttx/blob/master/boards/arm/stm32/stm32f4discovery/src/stm32_bringup.c#L140

So it doesn't matter which path you take to do that initialization, the board initialization is always performed in the same way.  Otherwise, the board will be initialized on one way if it is initialized from the application but in a different with if initialized by board_late_initialize().

I suppose that I can imagine some scenario where you might want both initializations and have them doing something different, in general board support logic, the two initialization paths should should have identical results.

Reply:
do you mean, the NSH_HAVEMMCSD block in board_app_initialize() should be moved to bringup()?

Reply:
Yes.  Is there some reason why the initialization should be different when it is performed via the boardctl->board_app_initialize() path vs the board_late_initialize() path.  I don't think so.  I think that the initialiaztion they should be identical use either path.  Many applications will not not call boardctl() to initialize.  In fact NSH does call boardctl() but most do not.  In the latter case, we just enable CONFIG_BOARD_LATE_INITIALIZE and we get the same initialization without board_app_initialize() being called.

Two separate initialization paths.  Two different use cases.  But the resulting initialization should be the same as a general rule.

I can imagine some special-case initialization that uses both, but I don't know when that would be used.  All of the configurations in the NuttX repository expect the initialization to be identical in either path.